src/main/java/com/aws/greengrass/clientdevices/auth/iot/Certificate.java

```diff
@@ -28,6 +105,16 @@ public String getNamespace() {
 
     @Override
     public Map<String, DeviceAttribute> getDeviceAttributes() {
-        return Collections.singletonMap("CertificateId", new StringLiteralAttribute(getIotCertificateId()));
+        return Collections.singletonMap("CertificateId", new StringLiteralAttribute(getCertificateId()));
+    }
+
+    private static String computeCertificateId(String certificatePem)
+            throws CertificateException, NoSuchAlgorithmException, UnsupportedEncodingException {
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        X509Certificate cert = (X509Certificate) cf.generateCertificate(
+                new ByteArrayInputStream(certificatePem.getBytes(StandardCharsets.UTF_8)));
+        MessageDigest digest = MessageDigest.getInstance("SHA-256");
+        digest.reset();
+        return new String(Hex.encode(digest.digest(cert.getEncoded())), "UTF-8");
```

Comment:
`Digest.calculate()` from Nucleus does exactly this. Can we reuse that?

Reply:
It doesn't though. That base64 encodes. This hex encodes